src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -139,15 +139,14 @@ public override void Initialize(AnalysisContext context)
                     return;
                 }
 
-                var typeName = WellKnownTypeNames.SystemOperatingSystem;
-
-                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(typeName, out var operatingSystemType))
+                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemOperatingSystem, out var operatingSystemType))
                 {
                     return;
                 }
 
                 var osPlatformType = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesOSPlatform);
                 var runtimeInformationType = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesRuntimeInformation);
+                var mockOSType = context.Compilation.GetOrCreateTypeByMetadataName("System.MockOperatingSystem");
```

Comment:
Is `mockOSType` (`System.MockOperatingSystem`) needed only for allowing the tests to override the annotations on `System.OperatingSystem`? If so, we shouldn't keep this code in here; instead, we could request to have the runtime updated in this repo to have the annotated `System.OperatingSystem` and comment out the tests that need them until that update occurs.

Reply:
> Is mockOSType (System.MockOperatingSystem) needed only for allowing the tests to override the annotations on System.OperatingSystem

Yes and `OperatingSystem.IsMacCatalys()` is also not available, we used mock attributes/APIs for the previous versions until the repo could use new APIs. Anyway, moved the mock API and tests into Debug mode, hope the CI would pass